File: backend-laravel/app/Models/Car.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Car extends Model {
    use HasFactory;

    protected $fillable = [
        'name', 'model', 'plate_number', 'daily_rate', 'status', 'last_oil_change', 'oil_change_km'
    ];

    public function bookings() {
        return $this->hasMany(Booking::class);
    }

    public function expenses() {
        return $this->hasMany(Expense::class);
    }
}

File: frontend-flutter/lib/models/car.dart
class Car {
  final int id;
  final String name;
  final String model;
  final String plateNumber;
  final double dailyRate;
  final String status;

  Car({
    required this.id,
    required this.name,
    required this.model,
    required this.plateNumber,
    required this.dailyRate,
    required this.status,
  });

  factory Car.fromJson(Map<String, dynamic> json) {
    return Car(
      id: json['id'],
      name: json['name'],
      model: json['model'],
      plateNumber: json['plate_number'],
      dailyRate: double.parse(json['daily_rate'].toString()),
      status: json['status'],
    );
  }
}
